src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -110,6 +110,11 @@ def getTasks(env)
     end
   end
 
+  def getTaskData(environment_name, module_name, task_name)
+    qualified_task_name = task_name == 'init' ? module_name : "#{module_name}::#{task_name}"
```

Comment:
This line's a bit hard to read, I was thinking maybe some parens around the equality comparison might help? Or a question mark in the assignment variable (`qualified_task_name?`)
And possibly a comment explaining this interface, since it's a little hard to figure out why it is the way it is